### ISSUE TITLE
Kudu Connector Can Not AutoCreate Table from Sink Record

### ISSUE THREAD

**Issue by rad08d** [Landoop/stream-reactor#497]
After POSTing a connector config like the following:
```
{
  "name": "troublesome-connector",
  "config": {
    "connector.class": "com.datamountaineer.streamreactor.connect.kudu.sink.KuduSinkConnector",
    "connect.kudu.schema.registry.url": "http://schema-registry:8081",
    "connect.kudu.kcql": "UPSERT INTO impala::default.troublesome-connector SELECT * FROM troublesome-connector AUTOCREATE DISTRIBUTEBY key_id INTO 5 BUCKETS AUTOEVOLVE",
    "schema.compatibility": "FULL",
    "topics": "troublesome-connector",
    "tasks.max": "1",
    "name": "troublesome-connector",
    "connect.kudu.master": "kudu-master:7051",
    "auto.create": "true"
  }
}
```
and putting a record on Kafka, the Kudu connector fails to auto-create the table from the sink record (specifically, the `DBHandlercreateTableFromSinkRecord` method). I believe this is due to a lack of a primary key. The following stacktrace ensues: 
```
connect_1          | [2018-09-13 21:16:20,297] ERROR Encountered error must specify at least one key column (com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter)
connect_1          | org.apache.kudu.client.NonRecoverableException: must specify at least one key column
connect_1          | 	at org.apache.kudu.client.KuduException.transformException(KuduException.java:110)
connect_1          | 	at org.apache.kudu.client.KuduClient.joinAndHandleException(KuduClient.java:352)
connect_1          | 	at org.apache.kudu.client.KuduClient.createTable(KuduClient.java:96)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.DbHandler$.executeCreateTable(DbHandler.scala:334)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.DbHandler$.createTableFromSinkRecord(DbHandler.scala:309)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter.applyDDLs(KuduWriter.scala:156)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter.com$datamountaineer$streamreactor$connect$kudu$sink$KuduWriter$$handleSinkRecord$1(KuduWriter.scala:129)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(KuduWriter.scala:138)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(KuduWriter.scala:138)
connect_1          | 	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
connect_1          | 	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
connect_1          | 	at scala.collection.Iterator$GroupedIterator.takeDestructively(Iterator.scala:1074)
connect_1          | 	at scala.collection.Iterator$GroupedIterator.go(Iterator.scala:1089)
connect_1          | 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1126)
connect_1          | 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1130)
connect_1          | 	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
connect_1          | 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter$$anonfun$1.apply$mcV$sp(KuduWriter.scala:141)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter$$anonfun$1.apply(KuduWriter.scala:141)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter$$anonfun$1.apply(KuduWriter.scala:141)
connect_1          | 	at scala.util.Try$.apply(Try.scala:192)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter.applyInsert(KuduWriter.scala:136)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduWriter.write(KuduWriter.scala:100)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduSinkTask$$anonfun$put$2.apply(KuduSinkTask.scala:67)
connect_1          | 	at com.datamountaineer.streamreactor.connect.kudu.sink.KuduSinkTask$$anonfun$put$2.apply(KuduSinkTask.scala:67)
```
I took a stab at fixing this issue and have a PR to be reviewed. Will attach it to this ticket after it is filled.

**Comment by rad08d** [comment]
PR: https://github.com/Landoop/stream-reactor/pull/498

**Comment by Antwnis** [comment]
@andrewstevenson can u check ?

**Comment by andrewstevenson** [comment]
Closed in #498
